control-plane/pkg/kafka/clientpool/clientpool.go

```diff
@@ -114,16 +123,16 @@ func (cp *ClientPool) GetClient(ctx context.Context, bootstrapServers []string,
 		}()
 	})
 
-	val.incrementCallers()
 	return val, nil
 }
 
 func (cp *ClientPool) GetClusterAdmin(ctx context.Context, bootstrapServers []string, secret *corev1.Secret) (sarama.ClusterAdmin, error) {
-	c, err := cp.GetClient(ctx, bootstrapServers, secret)
+	c, err := cp.getClient(ctx, bootstrapServers, secret)
 	if err != nil {
 		return nil, err
 	}
 
+	c.incrementCallers()
```

Comment:
isn't this incremented in `getClient`

Reply:
we could leave the usage of the public one to avoid duplication, so that the increment is done only in one place

Reply:
Not anymore, I removed the calls from there.

Now we explicitly increment it in the admin or client public methods

Reply:
I mean this 

```
	c, err := cp.getClient(ctx, bootstrapServers, secret)
	if err != nil {
		return nil, err
	}

	c.incrementCallers()
```

can be replaced with:

```
	c, err := cp.GetClient(ctx, bootstrapServers, secret)
	if err != nil {
		return nil, err
	}

```

Reply:
oh I see what you mean, let me change that